primitives/finality-grandpa/src/lib.rs

```diff
@@ -370,25 +373,32 @@ where
 /// Localizes the message to the given set and round and signs the payload.
 #[cfg(feature = "std")]
 pub fn sign_message<H, N>(
+	keystore: BareCryptoStorePtr,
 	message: grandpa::Message<H, N>,
-	pair: &AuthorityPair,
+	public: &AuthorityId,
```

Comment:
`public` gets cloned at the end anyway, so it makes more sense to pass ownership into the function.

Reply:
The suggestion here would try to move `local_id` from `OutgoingMessages`(`self` in this case). The communication mod does something like this:
```
if let Some(ref public) = self.local_id {
```

Reply:
Better explicit move than implicit clone IMO

Reply:
Sorry, do you mean i should move `local_id` out of `self`? if so, what would happen looks like:
```
error[E0507]: cannot move out of dereference of `std::pin::Pin<&mut communication::OutgoingMessages<Block>>`
   --> client/finality-grandpa/src/communication/mod.rs:670:25
    |
670 |         if let Some(public) = self.local_id {
    |                     ------    ^^^^^^^^^^^^^ help: consider borrowing here: `&self.local_id`
    |                     |
    |                     data moved here
    |                     move occurs because `public` has type `sp_finality_grandpa::app::Public`, which does not implement the `Copy` trait
```

Reply:
```rust
if let Some(ref public) = self.local_id {
   call_fn(public.clone())
}
```

Reply:
Got you... i committed this change.